src/hotspot/share/utilities/breakpoint.hpp

```diff
@@ -25,11 +25,52 @@
 #ifndef SHARE_UTILITIES_BREAKPOINT_HPP
 #define SHARE_UTILITIES_BREAKPOINT_HPP
 
-// If no more specific definition provided, default to calling a
-// function that is defined per-platform.  See also os::breakpoint().
+#include <csignal>
+
+// BREAKPOINT
+//
+// Programatically triggers a breakpoint for debuggers.
+
+#if defined(TARGET_COMPILER_gcc)
+
+#ifdef __has_builtin
+#if __has_builtin(__builtin_debugtrap)
+#define BREAKPOINT __builtin_debugtrap()
+#endif
+#endif
+
 #ifndef BREAKPOINT
-extern "C" void breakpoint();
-#define BREAKPOINT ::breakpoint()
+#define BREAKPOINT __builtin_trap()
+#endif
+
+#elif defined(TARGET_COMPILER_visCPP)
+
+#include <intrin.h>
+
+#pragma intrinsic(__debugbreak)
+
+#define BREAKPOINT __debugbreak()
+
+#elif defined(TARGET_COMPILER_xlc)
+
+#if defined(SIGTRAP)
+
+#define BREAKPOINT ::raise(SIGTRAP)
+
+#elif defined(SIGINT)
```

Comment:
The VM already uses SIGINT to shutdown the VM - raising it here doesn't seem right.